[Executorch][Quant][BE] Refactor Choose_Qparams

JacobSzwejbka opened the pull request:
Summary: Refactor so that it can be decomposed

Test Plan: ci

Differential Revision: D42681268

jerryzh168 commented on torch/ao/quantization/utils.py:
```diff
@@ -476,6 +477,105 @@ def _normalize_kwargs(func: Callable, loc: Dict[str, Any]) -> "OrderedDict[str,
+# Functionally equivalent to '_calculate_qparams' in observer.py. Observers must be torchscriptable however and qscheme
+# as far as I can tell is not allowed to passed as a parameter in torchscript functions. This makes refactoring observer
+# to use this utility a massive pain and very gross. For now Im opting just to duplicate as this code seems unlikey to change
+# (last update over 1 year ago) and when torchscript is fully deprecated we can refactor.
```
nit: maybe just add a TODO string here so that we can find it later

jerryzh168 commented on torch/ao/quantization/observer.py:
```diff
@@ -292,6 +292,11 @@ def _validate_qmin_qmax(self, quant_min: int, quant_max: int) -> None:
+    # Functionally equivalent to 'determine_qparams' in utils.py. Observers must be torchscriptable however and qscheme
+    # as far as I can tell is not allowed to passed as a parameter in torchscript functions. This makes refactoring observer
+    # to use this utility a massive pain and very gross. For now Im opting just to duplicate as this code seems unlikey to change
+    # (last update over 1 year ago) and when torchscript is fully deprecated we can refactor.
```
same here

jerryzh168 reviewed (approved):
looks good, thanks! please fix the lint errors before landing as well

JacobSzwejbka committed:
[Executorch][Quant][BE] Refactor Choose_Qparams (#94338)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/94338

Refactor so that it can be decomposed

Test Plan: ci

Reviewed By: jerryzh168

Differential Revision: D42681268

fbshipit-source-id: 072fd62ea40a8b7134733a994ccd9e6f04063b1e

JacobSzwejbka commented:
@pytorchbot merge -g
